### PR TITLE
Added steps for setting up IAM, Lambda, API Gateway

### PR DESCRIPTION
**Pull request by RyanRussell00**
Added step-by-step instructions on reproducing the IAM, Lambda and API Gateway used for the crowfacts project.

Still need to do:
- How to connect your front-end to API Gateway
- What is CORS and how to use it

### CODE REVIEW

File: _docs/crowfacts.md [hunk 1]
```diff
@@ -7,6 +7,160 @@ permalink: /crowfacts.html
...
+**Hint: If you are using Python you may get an error saying "inconsistent use of tabs and spaces...". In this case just go back to your code and replace all the indentations with tabs.**
+
+If you see the item you are expecting in the "Execution result" banner at the top, then you've successfully written your first lambda function!!
+
+## Step 4: Connecting Lambda to API Gateway
+
+1. In order to connect our Lambda to API Gateway we need to create an Alias for our Lambda. Open up your Lambda function and select "Actions" at the top and then select "Create alias".
+Give you alias an easy to spell name, in this case we will use "Latest". Ignore the description. Under "Version" make sure to select "$LATEST". Ignore the "Additional version".
+
+2. Head over to [API Gateway service](https://us-west-2.console.aws.amazon.com/apigateway/main/apis?region=us-west-2).
```
Comment (etcadinfinitum) [inline _docs/crowfacts.md:134]: Nope, do not put the console URL in here -- for one, not everybody will have an AWS account to follow along, and secondly, not everyone will be using the US-Oregon region, and creating different product instances across regions can have unintended negative consequences (like being unable to communicate between services across regions).

File: _docs/crowfacts.md [hunk 2]
```diff
@@ -7,6 +7,160 @@ permalink: /crowfacts.html
...
+- Lambda
+- IAM
 
-More information coming soon!
+# Connecting the Front-End to the Back-End
+Services used: IAM, Lambda, API Gateway
 
+## Step 1: Setting up IAM permissions
+
+1. Go to the [IAM services](https://console.aws.amazon.com/iam/home?region=us-west-2#/home) page and select "roles" on the left menu, and click on the blue "Create role" button.
```
Comment (etcadinfinitum) [inline _docs/crowfacts.md:22]: Nope, do not put the console URL in here -- for one, not everybody will have an AWS account to follow along, and secondly, not everyone will be using the US-Oregon region, and creating different product instances across regions can have unintended negative consequences (like being unable to communicate between services across regions).

File: _docs/crowfacts.md [hunk 3]
```diff
@@ -7,6 +7,160 @@ permalink: /crowfacts.html
...
+## Step 3: Make your Lambda useful (with code!)
+
+1. Here is the code we use for retrieving a single item from the database. Please take the time to read the comments in the code as they explain what the code does.
+
+**HINT:** Most of the code can be reused to accomodate for PUT, POST, GET, etc. calls.
+
+**HINT 2:** Make sure you are not using reserved keywords in your code! [These](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/ReservedWords.html) are all the reserved keywords for DynamoDB.
+
+
+```
```
Comment (etcadinfinitum) [inline _docs/crowfacts.md:50]: Recommend _not_ putting code in here; that's what the [CrowFacts repository](https://github.com/UWB-ACM/crowfacts) is for.

If you _must_ put code in here, make sure to do syntax highlighting, lest the code be a block of monocolored text which doesn't look like code.

````suggestion
```python3
````

File: _docs/crowfacts.md [hunk 4]
```diff
@@ -7,6 +7,160 @@ permalink: /crowfacts.html
...
+    
+	# Check that the incoming request has the necessary parameters. In this case we are expecting 'CrowSpecies' and 'location'. 
+	# For a GET method, the incoming parameters need to be the primary/sorting keys of the database.
+	# For a POST method, the incoming parameters need to be the new data you want to add to the database
+    
+	try:
+        species = event['CrowSpecies']
+        loc = event['location']
+    except Exception as e:
+		return e.response['Error']['Message']
```
Comment (etcadinfinitum) [inline _docs/crowfacts.md:64]: Indentation here looks visually wrong -- Python is very particular about this. If it wouldn't run in production, we shouldn't put in in our docs.

File: _docs/crowfacts.md [hunk 5]
```diff
@@ -7,6 +7,160 @@ permalink: /crowfacts.html
...
+		response = table.get_item(
+			
+			# The Key is what we use to query the database (ie: We need the key to know what item to return)
+			
+			Key={
+				'CrowSpecies': species,
+				'location': loc
+			}
+		)
+	except Exception as e:
```
Comment (etcadinfinitum) [inline _docs/crowfacts.md:87]: Why so much whitespace? This becomes hard to read. It's a pretty particular thing, but there's no good reason for whitespace to be between a comment and the code the comment is referring to, at a bare minimum, imo. If I were you, I'd scrap all unnecessary line breaks.

Comment (etcadinfinitum) [inline _docs/crowfacts.md:87]: Generic exceptions are always suspicious. Recommend [reading this, and all the way to the bottom.](https://github.com/boto/boto3/issues/1262#issuecomment-467465203)

**Review by etcadinfinitum — CHANGES_REQUESTED** [review]
General comments:

- Much of this content is written as if the reader is going to recreate this project in its entirety, instead of generalizing the problems we encountered and using the takeaways & resources to work on a different project. **We want to provide insight about how the project is structured; we don't need a step-by-step set of instructions.**
- Step by step processes for setting up permissions (and probably other things) should be directed to an official tutorial. This will help generalize _this_ project for a wider audience.
- If you aren't directing to official tutorials or AWS docs, at a minimum, you should include screenshots. A text description ("the big orange button in the corner") isn't nearly as useful as visualizing where the GUI object is that you're interested in. _It also isn't as compelling for the average reader._

This is a surprising amount of detail, and seems unnecessary considering the volume of literature that's out on the internet about using AWS, Dynamo, API Gateway, and Lambda. Think of a water balloon of information, not a dunk tank.

I recommend looking at the feelgood page and Anish's [PR](https://github.com/UWB-ACM/Hackathon-Docs-2020/pull/31) in detail to get more of an idea of what information is detailed step by step versus what concepts are named and then linked externally.

**Comment by RyanRussell00** [comment]
@etcadinfinitum Thanks for the review, I will go ahead and fix it and resubmit the PR.

**Pull request closed without merging**